string.c

```diff
@@ -9662,24 +9653,10 @@ rb_str_chomp_bang(int argc, VALUE *argv, VALUE str)
 
 /*
  *  call-seq:
- *     str.chomp(separator=$/)   -> new_str
- *
- *  Returns a new String with the given record separator removed
- *  from the end of <i>str</i> (if present). If <code>$/</code> has not been
- *  changed from the default Ruby record separator, then <code>chomp</code> also
- *  removes carriage return characters (that is, it will remove <code>\n</code>,
- *  <code>\r</code>, and <code>\r\n</code>). If <code>$/</code> is an empty string,
- *  it will remove all trailing newlines from the string.
- *
- *     "hello".chomp                #=> "hello"
- *     "hello\n".chomp              #=> "hello"
- *     "hello\r\n".chomp            #=> "hello"
- *     "hello\n\r".chomp            #=> "hello\n"
- *     "hello\r".chomp              #=> "hello"
- *     "hello \n there".chomp       #=> "hello \n there"
- *     "hello".chomp("llo")         #=> "he"
- *     "hello\r\n\r\n".chomp('')    #=> "hello"
- *     "hello\r\n\r\r\n".chomp('')  #=> "hello\r\n\r"
+ *    chomp(line_sep = $/) -> new_string
```

Comment:
I'm not sure if `line_sep` is a better name for this, since it could be something that's not a "line separator" in a traditional sense (to me, line separator means `"\n"` or `"\r\n"`).

E.g. in this example, `"o"` is not really a "line separator":

```ruby
irb(main):009:0> "helloo".chomp("o")
=> "hello"
```

Reply:
$/, the default, is called in globals.rdoc "input record separator" and module English calls it "$INPUT_RECORD_SEPARATOR".  At String#lines, I called it line_sep, not record_sep, following the method name.  Almost always, chop and chomp are about lines.  So this seems useful to me.

Reply:
That's fair.